test/e2e/logs_test.go

```diff
@@ -66,5 +66,10 @@ func retrieveLog(nodeName, srcPath, destDir string) error {
 		return errors.Wrap(err, "failed to create log directory")
 	}
 	outputFile := filepath.Join(destDir, splitPath[0], filepath.Base(srcPath))
-	return ioutil.WriteFile(outputFile, out, os.ModePerm)
+	ioutil.WriteFile(outputFile, out, os.ModePerm)
+	logInfo, err := os.Stat(outputFile)
+	if err != nil || logInfo.Size() == 0 {
+		return errors.Wrap(err, "log file should not be empty")
+	}
```

Comment:
I feel like this should be done at the retrieveLogs() call site to avoid giving too much responsibility to this function. Also we're retrying on error at the call site -- when we get a log size of zero, we don't need to retry